feat(*): add examples icons

dmitrsavk opened the pull request:
## Новый компонент `<Icon />`

Процесс публикации иконок следующий:

1. Дизайнер рисует иконку в фигме, дает ей корректное название.

2. С помощью нашего фигма-плагина создается пулл-реквест с иконкой в репозиторий alfa-ui-primitives.

3. После мерджа пулл-реквеста в alfa-ui-primitives, начинается процесс генерации реакт-компонента на основе svg-файла иконки.

4. После успешной генерации реакт-компонента, публикуется новая версия пакета.

Все иконки в alfa-ui-primitives имеют префикс, соответствующий названию набора иконок.
**По префиксу иконки будут разделены по npm-пакетам.**

Сейчас почти у всех иконок префикс `icon_` — это легаси-префикс. Новые наборы иконок будут иметь другие префиксы. Например, Клик и Альфа-Мобайл объединяют свои иконки и предложили название «glyph»: у иконок будет префикс `glyph_` и они попадут в соответствующий npm-пакет.

**Примеры**

1. `website_arrow-down_m.svg` → `ArrowDownMIcon.tsx`
    Будет доступна в npm-пакете `@alfalab/icons-website`

    `import ArrowDownMIcon from '@alfalab/icons-website/ArrowDownMIcon'`
    `<ArrowDownMIcon />`

2. `glyph_chevron-up_m.svg` → `ChevronUpMIcon.tsx`
    Будет доступна в npm-пакете `@alfalab/icons-glyph`

    `import ChevronUpMIcon from '@alfalab/icons-website/ChevronUpMIcon'`
    `<ChevronUpMIcon />`

**_В этом реквесте показан пример на тестовых иконках из assets-temp._**

dmitrsavk committed:
feat(*): add examples icons

reme3d2y commented:
Я бы все-таки добавил ...rest в пропсы. Хуже от этого не будет, зато остается возможность что-то навесить, если вдруг понадобится.

dmitrsavk committed:
chore(*): add building icons pakcages

SiebenSieben commented on packages/icon/package.json:
```diff
@@ -0,0 +1,10 @@
+{
+    "name": "@alfalab/icons-icon",
```
Может зашить для этого набора другое название?

А то поменять `icon` пока не можем и кривое имя получается )

Вариант: `icons-classic`

dmitrsavk replied:
Сейчас поменяю

etroynov replied:
а почему не просто icons?

SiebenSieben replied:
По префиксу делим все иконки на несколько npm-пакетов с названиями icons-…

Если этот назвать просто icons, то будет странно, другие пакеты ведь тоже с иконками

dmitrsavk committed:
chore(*): remove 'class' svg attr

dmitrsavk committed:
chore(*): rename package 'icon' to 'classic'

etroynov commented:
> Я бы все-таки добавил ...rest в пропсы. Хуже от этого не будет, зато остается возможность что-то навесить, если вдруг понадобится.

будет хуже и еще как, найдется человек который пихнет туда большой объект

dmitrsavk committed:
chore(*): remove exrta tsconfig

dmitrsavk committed:
chore(*): fix version

dmitrsavk committed:
Publish

 - @alfalab/icons-classic@0.0.2
 - @alfalab/icons-click@0.0.2
 - @alfalab/icons-website@0.0.2

dmitrsavk committed:
Publish

 - @alfalab/icons-classic@0.0.3
 - @alfalab/icons-click@0.0.3
 - @alfalab/icons-website@0.0.3

dmitrsavk committed:
chore(*): update configs
